### PR TITLE
Add `submit-review-as-single-comment` feature

### LINKED ISSUE #2063: Submit pending review comment as "single comment"
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/refined-github/issues/2063)

1. Answer a review comment, like https://github.com/sindresorhus/refined-github/pull/1271/files#r183043274
3. Press "Start review" instead of "Add single comment"
4. Be stuck with a pending answer that's not supposed to be a whole new review

Screenshot 2019-05-21 at 21 02 57

Sometimes I just want to leave a single comment. It's almost ok if it's a new comment (I can just press "Submit review" as comment), but if you're answering an existing thread, it will look strange on the Conversation tab, disconnected from the original comment.

I wish that clicking "Pending" would send the comment instantly (after `confirm()`). I'm not sure if this is feasible because you probably have to reuse another autenticity token from the page.

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $60.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($60.00)
### Submitted pull Requests
- [#2188 Add `submit-review-as-single-comment` feature](https://issuehunt.io/r/sindresorhus/refined-github/pull/2188)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/refined-github/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by bfred-it**
Closes #2063

# Testing

This has to support various situations:

- Multiple comment boxes open
- Unified or split diffs
- Multiple threads on the same line
- New thread or reply to an existing thread

### Example of what it should support

Screenshot

## Helpful code for testing

```js
// This will delete all the review comments on a PR
$$('[data-confirm="Are you sure you want to delete this?"]').map(el => {delete el.dataset.confirm; el.click()} )
```

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#2063: Submit pending review comment as "single comment"](https://issuehunt.io/repos/51769689/issues/2063)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

File: source/content.ts [hunk 1]
```diff
@@ -107,6 +107,7 @@ import './features/highest-rated-comment';
 import './features/clean-issue-filters';
 import './features/minimize-upload-bar';
 import './features/cycle-lists-with-keyboard-shortcuts';
+import './features/submit-review-as-single-comment';
```
Comment (bfred-it) [inline source/content.ts]: Needs a better name

File: readme.md [hunk 2]
```diff
@@ -105,6 +105,7 @@ GitHub Enterprise is also supported. More info in the options.
 - [Search for issues and PRs separately in the top search.](https://user-images.githubusercontent.com/1402241/52181103-35a09f80-2829-11e9-9c6f-57f2e08fc5b2.png)
 - [View the source of Markdown files.](https://user-images.githubusercontent.com/1402241/54814836-7bc39c80-4ccb-11e9-8996-9ecf4f6036cb.png)
 - [Update a PR from the base branch to ensure it build correctly before merging the PR itself (same-repo branches only).](https://user-images.githubusercontent.com/1402241/57941992-f2170080-7902-11e9-8f8a-594aad983559.png)
+- [Submit a single PR comment if you mistakenly started a new review.](https://user-images.githubusercontent.com/1402241/60331761-f6394200-99c7-11e9-81c2-c671cba9602a.gif)
```
Comment (bfred-it) [inline readme.md]: Ready!

**Comment by bfred-it** [comment]
I updated the behavior to **first** show the new field with the comment **and then** try to delete it. This is less visually clean but safer because the comment is always on screen, should the operation go wrong. 



Technically we can still hide the new field and show it only if there are issues. Thoughts?

File: package.json [hunk 3]
```diff
@@ -96,6 +96,7 @@
 			"__featureName__"
 		],
 		"rules": {
+			"no-alert": "off",
```
Comment (bfred-it) [inline package.json]: Comment test

**Comment by sindresorhus** [comment]
> Technically we can still hide the new field and show it only if there are issues. Thoughts?

I would hide it. Currently, it feels a bit slow and jumpy. It's like 3 slow steps and it takes 3 seconds for me. Not a big issue, but could be smoother.

**Comment by sindresorhus** [comment]
This works great though and it's super useful. I click the green Review button by accident almost every day.

**Comment by bfred-it** [comment]
> I would hide it. Currently, it feels a bit slow and jumpy


Time isn't affected, we still need to wait for the original comment to be deleted before submitting a new one, otherwise we risk them being submitted at once (behavior described in https://github.com/sindresorhus/refined-github/issues/2063#issuecomment-506437947)

**Comment by sindresorhus** [comment]
Ok. I would still hide it though.

**Pull request merged**